jupyterhub/spawner.py

```diff
@@ -652,21 +652,19 @@ def poll_and_notify(self):
                 self.log.exception("Unhandled error in poll callback for %s", self)
         return status
 
-    death_interval = Float(DT_MIN)
-
     @gen.coroutine
     def wait_for_death(self, timeout=10):
         """Wait for the single-user server to die, up to timeout seconds"""
-        loop = IOLoop.current()
-        tic = loop.time()
-        dt = self.death_interval
-        while dt > 0:
+        @gen.coroutine
+        def _wait_for_death():
             status = yield self.poll()
-            if status is not None:
-                break
-            else:
-                yield gen.sleep(dt)
-            dt = min(dt * DT_SCALE, DT_MAX, timeout - (loop.time() - tic))
+            return status is not None
+
+        yield exponential_backoff(
```

Comment:
I'd like to keep `start_wait=self.death_interval` so subclasses can slow this down if needed.